### PR TITLE
DOCSP-37469: Add transactions page

### PR DESCRIPTION
**Pull request by mcmorisi**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - <[DOCSP-37469](https://jira.mongodb.org/browse/DOCSP-37469)>
Staging - <https://preview-mongodbmcmorisi.gatsbyjs.io/csharp/DOCSP-37469-transactions-v2/fundamentals/transactions/>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/fundamentals/transactions.txt [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+   ``MongoDatabase`` or ``MongoCollection``) that created it. Using an
+   ``IClientSession`` with a different ``MongoClient`` results in operation
+   errors.
+
+Methods
+-------
+
+Create an ``IClientSession`` by using either the ``StartSession()`` or
+``StartSessionAsync()`` methods on your ``MongoClient`` instance, depending on your
+application's usecase. You can then modify the session state by using the method set
```
Comment (rustagir) [inline source/fundamentals/transactions.txt:57]:
```suggestion
Create an ``IClientSession`` by using either the synchronous ``StartSession()`` or
the asynchronous ``StartSessionAsync()`` method on your ``MongoClient`` instance. You can then modify the session state by using the method set
```

File: source/fundamentals/transactions.txt [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+provided by the ``IClientSession`` interface. Select from the following
+:guilabel:`Synchronous Methods` and :guilabel:`Asynchronous Methods`
+tabs to learn about the methods to manage your transaction that correspond to each
+framework:
```
Comment (rustagir) [inline source/fundamentals/transactions.txt:61]: S: maybe we can remove the end of the sentence as it is obvious
```suggestion
tabs to learn about the methods to manage your transaction:
```

File: source/fundamentals/transactions.txt [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+                |
+                | **Parameters**: ``Func <IClientSessionHandle, CancellationToken, Task<TResult>>``, ``TransactionOptions``, ``CancellationToken``
+                | **Return Type**: ``Task <TResult>``
+
+Example
+-------
+
+This example shows how you can create a session, create a
+transaction, and commit an insert operation into multiple collections through
+the following steps:
```
Comment (rustagir) [inline source/fundamentals/transactions.txt:185]:
```suggestion
This example shows how you can create a session, create a
transaction, and insert documents into multiple collections within the transaction
through the following steps:
```

File: source/fundamentals/transactions.txt [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+guide, see the following API Documentation:
+
+- `IClientSession <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.IClientSession.html>`__
+- `MongoClient <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.MongoClient.html>`__
+- `StartTransaction() <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.IClientSession.StartTransaction.html>`__
+- `AbortTransaction() <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.IClientSession.AbortTransaction.html>`__ / `AbortTransactionAsync() <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.IClientSession.AbortTransactionAsync.html>`__
+- `CommitTransaction() <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.IClientSession.CommitTransaction.html>`__ / `CommitTransactionAsync() <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.IClientSession.CommitTransactionAsync.html>`__
+- `WithTransaction() <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.IClientSession.WithTransaction.html>`__ / `WithTransactionAsync() <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.IClientSession.WithTransactionAsync.html>`__
+- `TransactionOptions() <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.TransactionOptions.html>`__
```
Comment (rustagir) [inline source/fundamentals/transactions.txt:228]:
```suggestion
- `TransactionOptions <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.TransactionOptions.html>`__
```

**Review by rustagir — APPROVED** [review]
approving with final comments to clean up some sentences!

File: source/fundamentals/transactions.txt [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+consistent, even if the operations encounter unexpected errors.
+
+When using the {+driver-short+}, you can create a new session from a
+``MongoClient`` instance as an ``IClientSession`` type. We recommend that you reuse
+your client for multiple sessions and transactions instead of
+instantiating a new client each time.
+
+.. warning::
+
+   Use an ``IClientSession`` only with the ``MongoClient`` (or associated
```
Comment (BorisDog) [inline source/fundamentals/transactions.txt:47]: Let's start using

File: source/fundamentals/transactions.txt [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+
+        .. list-table::
+            :widths: 40 60
+            :header-rows: 1
+
+            * - Method
+              - Description
+
+            * - ``StartTransaction()``
+              - | Starts a new transaction, configured with the given options, on
```
Comment (BorisDog) [inline source/fundamentals/transactions.txt:75]: Let's mention that `TransactionOptions` are optional here.

File: source/fundamentals/transactions.txt [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+                  this session. Returns an error if there is already
+                  a transaction in progress for the session. To learn more about
+                  this method, see the :manual:`startTransaction() page
+                  </reference/method/Session.startTransaction/>`  in the Server manual.
+                |
+                | **Parameter**: ``TransactionOptions``
+
+            * - ``AbortTransaction()``
```
Comment (BorisDog) [inline source/fundamentals/transactions.txt:83]: We don't return an error but throw an exception instead. This is applicable for all methods.

File: source/includes/fundamentals/code-examples/Transaction.cs [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+
+        // Cleans up the collections we'll be using
+        Setup(database);
+
+        // begin-transaction
+        var books = database.GetCollection<Book>("books");
+        var films = database.GetCollection<Film>("films");
+        
+        // Begins transaction
+        using (var session = mongoClient.StartSession()) {
```
Comment (BorisDog) [inline source/includes/fundamentals/code-examples/Transaction.cs:25]: Styling: C# usually uses Allman style braces
```
using (...)
{
}

try 
{
 ...
}
catch
{
}
```

**Review by BorisDog — COMMENTED** [review]
Looks good overall, added few comments.

**Review by BorisDog — APPROVED** [review]
LGTM!

**Pull request merged**